packages/modules/minio/src/minio-container.ts

```diff
@@ -0,0 +1,70 @@
+import { AbstractStartedContainer, GenericContainer, Wait, type StartedTestContainer } from "testcontainers";
+
+const MINIO_PORT = 9000;
+const MINIO_UI_PORT = 9001;
+
+export class MinioContainer extends GenericContainer {
+  private username = "minioadmin";
+  private password = "minioadmin";
+
+  constructor(image = "minio/minio:RELEASE.2024-12-13T22-19-12Z") {
+    super(image);
+    this.withExposedPorts(MINIO_PORT, MINIO_UI_PORT);
+    this.withWaitStrategy(Wait.forAll([Wait.forHttp("/minio/health/live", MINIO_PORT)]));
+    this.withCommand(["server", "--console-address", `:${MINIO_UI_PORT}`, "/data"]);
+  }
+
+  public withUsername(username: string): this {
+    this.username = username;
+    return this;
+  }
+
+  public withPassword(password: string): this {
+    this.password = password;
+    return this;
+  }
+
+  public override async start(): Promise<StartedMinioContainer> {
+    this.withEnvironment({
+      MINIO_ROOT_USER: this.username,
+      MINIO_ROOT_PASSWORD: this.password,
+    });
+    const startedContainer = await super.start();
+    return new StartedMinioContainer(startedContainer, this.username, this.password);
+  }
+}
+
+export class StartedMinioContainer extends AbstractStartedContainer {
+  private readonly port: number;
+  private readonly uiPort: number;
+
+  constructor(
+    startedTestContainer: StartedTestContainer,
+    private readonly username: string,
+    private readonly password: string
+  ) {
+    super(startedTestContainer);
+    this.port = startedTestContainer.getMappedPort(MINIO_PORT);
```

Comment:
These will need to be computed on demand to support the container restarting, as it restarts with different ports

Reply:
Is this something that has changed recently? I'm pretty sure there are existing modules with this pattern.

Reply:
Moved these to be resolved in `getPort`/`getUiPort` whenever they're called.

Reply:
Not all modules support restarting the container. You could see for yourself by adding a test for restarting the container if you'd like